### PR TITLE
Arion hs

### PR DESCRIPTION
**Pull request by roberth**
This is a feature branch with the goal of replacing the main script by a more maintainable and intelligent program. It was started during ZuriHac 2019.

TODO
 - [x] don't load images when using host store (bug in base image reimplementation)
 - [x] fix stderr streaming

### CODE REVIEW

File: live-check [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#!nix-shell ./shell.nix
+#!nix-shell -i bash
+set -eux -o pipefail
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+ghcid \
+  --command 'ghci -isrc/haskell/exe src/haskell/exe/Main.hs' \
+  --reload=src/haskell \
+  --restart=hercules-ci-api.cabal \
```
Comment (domenkozar) [inline live-check]: this should be arion.cabal

File: live-unit-tests [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#!nix-shell -i bash
+set -eux -o pipefail
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+ghcid \
+  --command 'cabal v2-repl arion-compose:arion-unit-tests --flags ghci --write-ghc-environment-files=never' \
+  --test=Main.main \
+  --reload=src/haskell \
+  --restart=hercules-ci-api.cabal \
```
Comment (domenkozar) [inline live-unit-tests]: same here

**Comment by roberth** [comment]
> I'm not a fan of maintaining a copy of docker-compose cli here, it's going to be a maintenance hell.

I agree we should keep the surface area small where possible.
Reasons why parts of the cli appear in the code:
 - The list of all commands needs to be there to decide whether to evaluate or build.
 - For exec I had to implement option parsing in order to extend its functionality.

> Why not have something like arion docker-compose <..> (and possibly arion dc as shortcut) which propagates everything to docker-compose?

As said above, knowing which commands need eval versus build is valuable. I don't see how such an indirection saves us maintenance. If some new command is introduced in docker compose, we're better off shielding that from users until arion can do the right thing for that command. I'd rather have a missing feature than a bug.
Note that arion doesn't have to work with any docker-compose version, but only the one that's packaged.

**Comment by domenkozar** [comment]
> Reasons why parts of the cli appear in the code:

OK, I guess that makes sense.

Shouldn't this PR also delete the bash bits?

**Pull request merged**